src/pass_manager.cpp

```diff
@@ -99,11 +98,18 @@ void run_passes(program& prog, const std::vector<pass>& passes, tracer trace)
     for(const auto& p : passes)
     {
         auto mods = prog.get_modules();
+        auto tree = prog.get_module_tree();
         for(const auto& mod : reverse(mods))
         {
             if(mod->bypass())
                 continue;
-            module_pm{mod, &prog, &trace}.run_pass(p);
+            module_pm mpm{mod, &trace};
+            mpm.prog     = &prog;
+            auto parents = range(tree.equal_range(mod));
+            // Assume only one parent for now
+            assert(distance(parents) < 2);
+            mpm.common_parent = (distance(parents) == 0) ? nullptr : parents.begin()->second;
```

Comment:
if distance(parents) == 0 then, mod must be a main_module, i guess. should we assert that ?

Reply:
No, it could also be an unused module.